### PR TITLE
refactor: Moves configuration creation for manifest to separate class

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

This changeset is intended to reduce the complexity of the `Client` class, which is currently 757 SLOC. This is partially achieved by extracting the logic to populate `Configuration` from the AndroidManifest to a separate class.

## Changeset

- Moves code which populates a configuration object from the AndroidManifest into a separate `ConfigFactory` class
- Update test imports to pass build

## Tests

Ran unit tests, mazerunner

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/Client.java [hunk 1]
```diff
@@ -258,94 +242,6 @@ public void update(Observable observable, Object arg) {
...
-
-    /**
-     * Populates the config with meta-data values supplied from the manifest as a Bundle.
-     *
-     * @param config the config to mutate
-     * @param data   the manifest bundle
-     * @return the updated config
-     */
-    @NonNull
-    static Configuration populateConfigFromManifest(@NonNull Configuration config,
```
Comment (martin308) [inline sdk/src/main/java/com/bugsnag/android/Client.java]: what's the default access modifier that's applied here? Does moving this have any external implications?

Comment (fractalwrench) [reply sdk/src/main/java/com/bugsnag/android/Client.java]: The package-level modifier will be applied, so anything with the `com.bugsnag.android` namespace can access this. I don't believe this has any external implications as I'm not aware of any other notifiers which call this method.
